binary_search_tree/binary_search_tree.py

```diff
@@ -5,10 +5,28 @@ def __init__(self, value):
     self.right = None
 
   def insert(self, value):
-    pass
+    cur = self
+    parent = None
+    while cur:
+      parent = cur
+      if value < cur.value:
+        cur = cur.left
+      else:
+        cur = cur.right
+    cur = BinarySearchTree(value)
 
```

Comment:
If cur is the last line in the function...what happens when there is no left or right node?